sysl2/sysl/cmd_export.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	sysl "github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/exporter"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type exportCmd struct {
+	appName string
+	out     string
+	mode    string
+	format  string
+}
+
+const (
+	swaggerMode = "swagger"
+	jsonMode    = "json"
+	yamlMode    = "yaml"
+)
+
+func (p *exportCmd) Name() string            { return "export" }
+func (p *exportCmd) RequireSyslModule() bool { return true }
+
+func (p *exportCmd) Configure(app *kingpin.Application) *kingpin.CmdClause {
+	cmd := app.Command(p.Name(), "Export sysl to external types. Supported types: Swagger")
+	cmd.Flag("app-name",
+		"name of the sysl app defined in sysl model."+
+			" if there are multiple apps defined in sysl model,"+
+			" swagger will be generated only for the given app").Default("all").Short('a').StringVar(&p.appName)
+	cmd.Flag("output",
+		"export filename in the format filename.export_format, "+
+			" supported options: filename.json; filename.yaml").Required().Short('o').StringVar(&p.out)
+	EnsureFlagsNonEmpty(cmd)
+	return cmd
+}
+
+func (p *exportCmd) writeSwaggerForApp(syslApp *sysl.Application, logger *logrus.Logger) error {
+	var output []byte
+	var err error
+	if p.mode == swaggerMode {
+		swaggerExporter := exporter.MakeSwaggerExporter(syslApp, logger)
+		err = swaggerExporter.GenerateSwagger()
+		if err != nil {
+			logger.Warnf("Error generating swagger for the application %s", err)
+			return err
+		}
+		output, err = swaggerExporter.SerializeOutput(p.format)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("unsupported export format")
+	}
+	filename := fmt.Sprintf("%s_%s.%s", strings.Split(p.out, ".")[0], syslApp.Name.Part[0], p.format)
+	fileWriteError := ioutil.WriteFile(filename, output, 0644)
```

Comment:
err := ioutil....

no need to keep adding new error vars unless you want to keep an old one for some reason

and you could just do `return ioutil.WriteFile()` in this case